src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
```

Comment:
Consider adding error handling for the 'get' method. If an invalid index is passed, the method will throw an IndexOutOfBoundsException, which is not handled here.